static/app/views/settings/organizationTeams/index.tsx

```diff
@@ -1,83 +1,82 @@
-import withOrganization from 'sentry/utils/withOrganization';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
+import type {AccessRequest} from 'sentry/types/organization';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
 
 import OrganizationTeams from './organizationTeams';
 
-type Props = {
-  api: Client;
-  organization: Organization;
-  teams: Team[];
-} & RouteComponentProps<{}, {}>;
+export function OrganizationTeamsContainer(props: RouteComponentProps<{}, {}>) {
+  const api = useApi();
+  const organization = useOrganization();
+  const [requestList, setRequestList] = useState<AccessRequest[]>([]);
 
-type State = DeprecatedAsyncView['state'] & {
-  requestList: AccessRequest[];
-};
-
-class OrganizationTeamsContainer extends DeprecatedAsyncView<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization} = this.props;
-
-    return [['requestList', `/organizations/${organization.slug}/access-requests/`]];
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    this.fetchStats();
-  }
+  const {isLoading, isError} = useApiQuery<AccessRequest[]>(
+    [`/organizations/${organization.slug}/access-requests/`],
+    {
+      staleTime: 0,
+      retry: false,
+      onSuccess: ([data]) => {
+        setRequestList(data);
+      },
+    }
+  );
 
-  fetchStats() {
-    const {organization} = this.props;
-    loadStats(this.props.api, {
+  useEffect(() => {
+    loadStats(api, {
       orgId: organization.slug,
       query: {
         since: (new Date().getTime() / 1000 - 3600 * 24).toString(),
         stat: 'generated',
         group: 'project',
       },
     });
-  }
+  }, [organization.slug, api]);
 
-  removeAccessRequest = (id: string, isApproved: boolean) => {
-    const requestToRemove = this.state.requestList.find(request => request.id === id);
-    this.setState(state => ({
-      requestList: state.requestList.filter(request => request.id !== id),
-    }));
-    if (isApproved && requestToRemove) {
-      const team = requestToRemove.team;
-      TeamStore.onUpdateSuccess(team.slug, {
-        ...team,
-        memberCount: team.memberCount + 1,
-      });
-    }
-  };
+  const handleRemoveAccessRequest = useCallback(
+    (id: string, isApproved: boolean) => {
+      const requestToRemove = requestList.find(request => request.id === id);
+      setRequestList(requestList.filter(request => request.id !== id));
+      if (isApproved && requestToRemove) {
+        const team = requestToRemove.team;
+        TeamStore.onUpdateSuccess(team.slug, {
+          ...team,
+          memberCount: team.memberCount + 1,
+        });
+      }
+    },
+    [requestList]
+  );
 
-  renderBody() {
-    const {organization} = this.props;
+  // can an organization be null here?
+  if (!organization) {
+    return null;
+  }
```

Comment:
For this not to break, you would need to pass `{allowNull: true}` to `useOrganization()`.